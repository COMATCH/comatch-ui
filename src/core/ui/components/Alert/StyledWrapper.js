import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { palette, typography } from '../../../styles/variables';

const FONT_COLORS = {
    info: palette.secondary,
    warning: palette.darkOrange,
    danger: palette.darkRed,
    success: palette.darkGreen,
};

const BACKGROUND_COLORS = {
    info: palette.lightBlue,
    warning: palette.lightOrange,
    danger: palette.lightRed,
    success: palette.lightGreen,
};

export const generateFontColor = ({ type }) => `color: ${FONT_COLORS[type]};`;

export const generateDsiplayStyling = ({ display, type, withIcon }) => {
    switch (display) {
        case 'box':
            return `
                background-color: ${BACKGROUND_COLORS[type]};
                padding: 15px;
                padding-bottom: 20px;

                ${withIcon ? 'padding-left: 40px;' : ''}

                .Alert__icon {
                    left: 15px;
                    top: 17px;
                }
            `;

        case 'notification':
            return `
                color: ${palette.white};
                background-color: ${BACKGROUND_COLORS[type]};
                display: inline-block;
                margin: 0;
                padding: 10px 10px 10px 35px;

                .Alert__icon {
                    left: 10px;
                    top: 13px;
                }

                .Alert__message {
                    font-weight: bold;
                }
            `;

        default:
            return '';
    }
};

export const StyledWrapper = styled(
    forwardRef(({ display, type, withIcon, ...rest }, ref) => (
        <section ref={ref} {...rest} />
    )),
)`
    border-radius: 3px;
    display: block;
    margin: 0 0 20px 0;
    padding: 0;
    padding-left: 25px;
    position: relative;
    // animation
    transition: opacity 500ms cubic-bezier(0.165, 0.84, 0.44, 1), transform 500ms cubic-bezier(0.165, 0.84, 0.44, 1);

    &.fade {
        transform: scale(1) translateZ(0);
        opacity: 1;
        will-change: transform, opacity;
    }

    &.slide-right {
        transform: translate3d(0, 0, 0);
        will-change: transform;
    }

    &.dismissed {
        &.fade {
            transform: scale(0.75) translateZ(0);
            opacity: 0;
        }

        &.slide-right {
            transform: translate3d(100%, 0, 0);
        }
    }

    // Common styles for elements
    .Alert__icon,
    .Alert__message {
        display: inline-block;
    }

    .Alert__icon {
        // font
        font-size: 15px;
        // positioning
        position: absolute;
        top: 2px;
        left: 0;
    }

    .Alert__message {
        font-size: ${typography.fontSm};
        margin: 0;
        padding: 0;
        line-height: ${typography.lineHeightSm};
    }

    ${generateFontColor}
    ${generateDsiplayStyling}
`;
