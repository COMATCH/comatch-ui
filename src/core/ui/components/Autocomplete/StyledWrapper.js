/* eslint-disable react/prop-types */
/* eslint-disable react/display-name */

/**
 * Reasoning behind disabling these rules:
 * - No need for the displayName on the Styled Wrapper
 * - No need for props validation since, they are already being validated
 *   within the Autocomplete component.
 */

import React, { forwardRef } from 'react';
import styled from 'styled-components';
import { inputs, palette, typography } from '../../../styles/variables';

const generateDisplayStyling = ({ display }) => {
    switch (display) {
        case 'block':
            return 'display: block;';
        default:
            return 'display: inline-block;';
    }
};

const generateDisabledStyling = ({ disabled }) => {
    if (!disabled) {
        return '';
    }

    return `color: ${palette.gray}`;
};

const generateHasErrorStyling = ({ hasError }) => {
    if (!hasError) {
        return '';
    }

    return `
        .SelectInput__input {
            box-shadow: 0 0 4px 0 rgba(173, 25, 25, 1), inset 0 0 2px 0 rgba(193, 193, 193, 0.5);
        }
    `;
};

const StyledWrapper = styled(
    forwardRef(({ display, disabled, hasError, ...rest }, ref) => <div {...rest} ref={ref} />),
)`
    ${generateDisplayStyling}
    font-size: ${typography.fontSm};
    margin-bottom: ${inputs.inputsMarginBottom};

    .Autocomplete__input {
        background-color: ${palette.white};
        border: 1px solid ${palette.midGray};
        border-radius: 3px;
        box-shadow: 0 0 4px 0 rgba(1, 156, 185, 0), inset 0 0 2px 0 rgba(193, 193, 193, 0.5);
        min-height: 40px;
        outline: 0 !important;
        transform: translateZ(0);
        transition: box-shadow 250ms ease-out;
        width: 100%;
        will-change: box-shadow;
        z-index: 3;

        &:focus {
            box-shadow: 0 0 4px 0 rgba(1, 156, 185, 1), inset 0 0 2px 0 rgba(193, 193, 193, 0.5);
        }
    }

    ${generateDisabledStyling}
    ${generateHasErrorStyling}
`;

export { StyledWrapper };
