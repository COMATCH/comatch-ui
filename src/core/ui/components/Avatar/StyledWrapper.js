import React, { forwardRef } from 'react';
import styled from 'styled-components';

function avatarSize({ diameter }) {
    const size = `${diameter >= 0 ? diameter : 45}px`;

    return `
        height: ${size};
        width: ${size};
    `;
}

const StyledTooltipWrapper = styled.div``;

const StyledBadgeWrapper = styled.div`
    align-content: center;
    background: transparent;
    display: flex;
`;

const Node = forwardRef(({ diameter, ...rest }, ref) => <div {...rest} ref={ref} />);
const StyledWrapper = styled(Node)`
    align-content: center;
    display: inline-flex;
    justify-content: center;
    position: relative;
    ${avatarSize}

    img {
        border-radius: 50%;
        height: 100%;
        overflow: hidden;
        width: 100%;
    }

    ${StyledBadgeWrapper} {
        bottom: -5px;
        position: absolute;
        right: -5px;
    }
`;

export { StyledBadgeWrapper, StyledTooltipWrapper, StyledWrapper };
