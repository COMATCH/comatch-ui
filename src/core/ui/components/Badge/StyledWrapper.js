import React from 'react';
import styled from 'styled-components';
import { palette, typography } from '../../../styles/variables';

export const getStylesBasedOnSize = ({ size }) => {
    switch (size) {
        case 'xs':
            return `
                border-radius: 9.5px;
                font-size: ${typography.fontXs};
                font-weight: bold;
                line-height: 1;
                padding: 3px 7px 5px;

                svg {
                    height: 10px;
                }
            `;

        case 'sm':
            return `
                border-radius: 15px;
                font-size: ${typography.fontSm};

                svg {
                    height: 12px;
                }
            `;

        default:
            return '';
    }
};

export const getColorBasedOnProps = ({ color }) => {
    switch (color) {
        case 'green':
            return `
                background-color: ${palette.green};
                border-color: ${palette.green};
                color: ${palette.white};
            `;

        case 'orange':
            return `
                background-color: ${palette.orange};
                border-color: ${palette.orange};
                color: ${palette.white};
            `;

        case 'gray':
            return `
                background-color: ${palette.gray};
                border-color: ${palette.gray};
                color: ${palette.white};
            `;

        case 'darkGray':
            return `
                background-color: ${palette.darkGray};
                border-color: ${palette.darkGray};
                color: ${palette.white};
            `;

        case 'white':
            return `
                background-color: ${palette.white};
                border-color: ${palette.gray};
                color: ${palette.gray};
            `;

        case 'blue':
            return `
                background-color: ${palette.secondary};
                border-color: ${palette.secondary};
                color: ${palette.white};
            `;

        case 'lightblue':
            return `
                background-color: ${palette.primary};
                border-color: ${palette.primary};
                color: ${palette.white};
            `;

        default:
            return '';
    }
};

export const StyledWrapper = styled(({ size, color, ...rest }) => <div {...rest} />)`
    border: 1px solid;
    display: inline-block;
    padding: 5px 8px;

    svg {
        margin-right: 5px;
        cursor: pointer;
    }

    ${getColorBasedOnProps}
    ${getStylesBasedOnSize}
`;
