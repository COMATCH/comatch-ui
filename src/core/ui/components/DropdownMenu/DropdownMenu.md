## DropdownMenu

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**className** | `String` | `null` | :x: | Additional class names
**id** | `String` | `null` | :x: | id of the container
**itemClassName** | `String` | `null` | :x: | Additional class names to be applied on each item
**items** | `Array[]<ReactNode>` | `[]` | :x: | The menu items

