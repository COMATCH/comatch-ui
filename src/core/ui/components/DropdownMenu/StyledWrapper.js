import styled from 'styled-components';
import { palette } from '../../../styles/variables';

export const ItemStyledWrapper = styled.div`
    background-color: transparent;
    color: ${palette.black};
    padding: 5px 10px;
    transition: background-color 250ms ease-out;

    &:hover {
        background-color: ${palette.lightGray};
    }
`;

export const StyledWrapper = styled.div`
    background-color: ${palette.white};
    border-radius: 5px;
    box-shadow: 0 2px 10px 0 rgba(0, 0, 0, 0.5);
    overflow: hidden;
`;
