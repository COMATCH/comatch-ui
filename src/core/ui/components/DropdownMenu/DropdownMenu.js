import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { Navigation, ORIENTATION } from '../Navigation';
import { StyledWrapper, ItemStyledWrapper } from './StyledWrapper';

const propTypes = {
    /**
     * Additional class names
     */
    className: PropTypes.string,
    /**
     * id of the container
     */
    id: PropTypes.string,
    /**
     * Additional class names to be applied on each item
     */
    itemClassName: PropTypes.string,
    /**
     * The menu items
     */
    items: PropTypes.arrayOf(PropTypes.node),
};

const defaultProps = {
    className: null,
    id: null,
    itemClassName: null,
    items: [],
};

export const DropdownMenu = ({ className, id, itemClassName, items }) => (
    <StyledWrapper {...id && { id }} className={classNames('DropdownMenu', className)}>
        <Navigation
            orientation={ORIENTATION.VERTICAL}
            items={items.map((item, index) => (
                <ItemStyledWrapper key={index} className={classNames('DropdownMenu__item', itemClassName)}>
                    {item}
                </ItemStyledWrapper>
            ))}
        />
    </StyledWrapper>
);

DropdownMenu.propTypes = propTypes;

DropdownMenu.defaultProps = defaultProps;
