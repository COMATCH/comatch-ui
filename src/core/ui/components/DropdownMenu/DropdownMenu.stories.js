import React from 'react';
import Highlight from 'react-highlight';
import { storiesOf } from '@storybook/react';
import { DropdownMenu } from './DropdownMenu';

// Can be HTML nodes or plain text
const dropdownMenuItems = ['Item 1', 'Item 2', 'Item 3', 'Item 4', 'Item 5'];
const dropdownMenuItemsString = `[${dropdownMenuItems.map((item) => `"${item}"`).join(',')}]`;

storiesOf('DropdownMenu', module)
    .add('with an empty list of items', () => (
        <>
            <DropdownMenu />
            <Highlight className="html">{`<DropdownMenu />`}</Highlight>
        </>
    ))
    .add('with basic list of items', () => (
        <>
            <DropdownMenu items={dropdownMenuItems} />
            <Highlight className="html">{`<DropdownMenu items={${dropdownMenuItemsString}} />`}</Highlight>
        </>
    ));
