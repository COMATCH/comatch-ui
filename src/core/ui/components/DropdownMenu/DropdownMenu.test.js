import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import '../../../config/tests/setup';

import classNames from 'classnames';
import { Navigation } from '../Navigation';
import { DropdownMenu } from './DropdownMenu';
import { ItemStyledWrapper } from './StyledWrapper';

describe('DropdownMenu', () => {
    const OPTIONAL_PROPS = {
        id: 'test',
        className: {
            test: true,
            skippedClassName: false,
        },
        itemClassName: 'ItemClassName',
        items: [
            'item 1',
            <span key="item-2" id="item-2">
                item 2
            </span>,
            'item 3',
        ],
    };

    it('should render correctly', () => {
        const dropdownMenu = renderer.create(<DropdownMenu />).toJSON();
        expect(dropdownMenu).toMatchSnapshot();
    });

    it('should render with default props', () => {
        const dropdownMenu = mount(<DropdownMenu />);

        expect(dropdownMenu.prop('id')).toEqual(null);
        expect(dropdownMenu.prop('className')).toEqual(null);
        expect(dropdownMenu.prop('itemClassName')).toEqual(null);
        expect(dropdownMenu.prop('items')).toEqual([]);
    });

    it('should render with custom props', () => {
        const props = { ...OPTIONAL_PROPS };
        const dropdownMenu = mount(<DropdownMenu {...props} />);

        expect(dropdownMenu.prop('id')).toEqual(props.id);
        expect(dropdownMenu.prop('className')).toEqual(props.className);
        expect(dropdownMenu.prop('itemClassName')).toEqual(props.itemClassName);
        expect(dropdownMenu.prop('items')).toEqual(props.items);
    });

    describe('verifying the HTML structure', () => {
        const props = { ...OPTIONAL_PROPS };
        const dropdownMenu = mount(<DropdownMenu {...props} />);

        it('should render a Navigation component', () => {
            expect(dropdownMenu.contains(Navigation)).toBeTruthy();
        });

        it('should render each item from the passed array', () => {
            const dropdownItems = dropdownMenu.find(ItemStyledWrapper);

            expect(dropdownItems).toHaveLength(props.items.length);
            dropdownItems.forEach((node) => {
                expect(node.hasClass(classNames(props.itemClassName))).toBeTruthy();
            });
        });
    });
});
