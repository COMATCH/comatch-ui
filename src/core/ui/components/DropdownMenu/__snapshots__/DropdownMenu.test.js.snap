// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu should render correctly 1`] = `
<div
  className="DropdownMenu sc-htpNat jgCLMM"
>
  <nav
    className="Navigation vertical sc-bdVaJa eeaaGT"
    orientation="vertical"
  >
    <ul />
  </nav>
</div>
`;
